lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -5,7 +5,6 @@ defmodule Kernel.WithTest do
 
   test "basic with" do
     assert with({:ok, res} <- ok(41), do: res) == 41
-    assert with(res <- four(), do: res + 10) == 14
```

Comment:
Was triggering a warning which I think is legit since this clause is always matching indeed:
<img width="679" alt="Screenshot 2024-01-30 at 8 59 44" src="https://github.com/elixir-lang/elixir/assets/11598866/f17e7420-23ef-4da2-ae8a-908d4c6149b5">

Maybe what could be improved could be to emit a different warning (e.g. using `<-` but this will always match, use `=` instead) during expansion and replace `<-` by a `=` in such a case, WDYT?

Reply:
@sabiwara :+1: to omit the else clause in this case and generate leaner code. :)

Reply:
Done in [e408364](https://github.com/elixir-lang/elixir/pull/13299/commits/e40836471a2b4be7d1dc8b67e4616505de0e5782)